virtcontainers/device/drivers/block.go

```diff
@@ -83,7 +83,17 @@ func (device *BlockDevice) Attach(devReceiver api.DeviceReceiver) (err error) {
 
 		switch customOptions["block-driver"] {
 		case "virtio-blk":
-			globalIdx = index
+
+			hypervisor := devReceiver.GetHypervisorType()
+
+			if hypervisor == "acrn" {
+				//With ACRN the rootfs for the VM itself
+				//sits at /dev/vda and consumes the first index.
+				globalIdx = index + 1
```

Comment:
In that case acrn should increase its `BlockIndex` when launching the guest, instead of hacking the index number on hotplug.

Reply:
globalIdx here is used to find the corresponding virtpath (/dev/vdX) starting with /dev/vda. Not sure how to get this information in ACRN which doesn't maintain the global mapping. This approach is very similar to the Firecracker approach. I believe this issue (https://github.com/kata-containers/runtime/issues/1061) should address this workaround as well. Let me know if I missed misunderstood.

Reply:
@vijaydhanraj `index` is maintained at the sandbox level and there are helper functions for ACRN to use. You can call `getAndSetSandboxBlockIndex()` to bump it once and for all when launching the guest ensuring that the first index is taken by the vm rootfs.

Reply:
ok thanks @bergwolf . I am bumping the index using getAndSetSandboxBlockIndex when acrn appendImage() is called and have left the globalIdx unmodified.